common/src/models/error.rs

```diff
@@ -80,6 +82,131 @@ impl ApiError {
     }
 }
 
+pub trait ErrorContext<T> {
```

Comment:
What do you think of having a generic `context_error(self, message: &str, status_code: StatusCode)` (and the corresponding `with` variant) instead of dedicated method each time ?

Reply:
Nah, then each caller would have type out the status code each time. Not a fan of passing `booleans` as args to functions and this seems close to the same thing.

Reply:
That's an idea, but I can abstract it more though. Ie I won't be able to make them use the generic one since they log different `warn!` and `error!`.

I mean they are the same in that they use a dynamic control (the args) when we know which concrete type we want when we write the code. So their dynamic nature at runtime feels the same.

Reply:
Ah, make sense. Ok let's keep it that way for now then, we'll see when the need arise.